src/awkward/_v2/forms/form.py

```diff
@@ -0,0 +1,63 @@
+# BSD 3-Clause License; see https://github.com/scikit-hep/awkward-1.0/blob/main/LICENSE
+
+from __future__ import absolute_import
+
+import json
+
+import awkward as ak
+
+np = ak.nplike.NumpyMetadata.instance()
+
+
+class Form(object):
+    @property
+    def has_identities(self):
+        return self._has_identities
+
+    @property
+    def parameters(self):
+        return self._parameters
+
+    @property
+    def form_key(self):
+        return self._form_key
+
+    def parameter(self, key):
+        if self._parameters is None:
+            return None
+        else:
+            return self._parameters.get(key)
+
+    def __str__(self):
+        return json.dumps(self.tolist(verbose=False), indent="    ")
+
+    def tolist(self, toplevel=False, verbose=True):
+        out = self._tolist_part(verbose=verbose)
+        if verbose:
+            out["has_identities"] = self._has_identities
+            out["parameters"] = self._parameters
+            out["form_key"] = self._form_key
+        else:
+            if self._has_identities:
+                out["has_identities"] = self._has_identities
+            if len(self._parameters) > 0:
+                out["parameters"] = self._parameters
+            if self._form_key is not None:
+                out["form_key"] = self._form_key
+        return out
```

Comment:
`toplevel` is not used in this function (and flake8 might be complaining about that).

The verbosity tests could have been implemented as

```python
            if verbose or self._has_identities:
                out["has_identities"] = self._has_identities
            if verbose or len(self._parameters) > 0:
                out["parameters"] = self._parameters
            if verbose or self._form_key is not None:
                out["form_key"] = self._form_key
```

with less duplication.